dev-requirements.txt

```diff
@@ -0,0 +1,10 @@
+pylint~=2.3
+flake8~=3.7
+isort~=4.3
+black>=19.3b0,==19.*
+mypy==0.740
+sphinx~=2.1
+sphinx-rtd-theme~=0.4
+sphinx-autodoc-typehints~=1.10.2
+pytest!=5.2.3
+pytest-cov>=2.8
```

Comment:
I understand these were moved from `tox.ini`, but what do you think about using the latest version of build tools instead of pinning these? It does mean dealing with some build unpredictability, but prevents us having to make big changes to update these packages later.

Reply:
I think we should at least have `>=` requirements (although the minimum version would be very low due to Python 3.4 support most of the time, at least for requirements we use on Python != 3.7). I'm fine with removing upper bounds though.